### ISSUE TITLE
False positive for W0602 when using `del`

### ISSUE THREAD

**Issue by KotlinIsland** [PyCQA/pylint#5333]
### Bug description
```py
a = 1


def foo():
    global a
    del a
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
Sandpit\a.py:5:4: W0602: Using global for 'a' but no assignment is done (global-variable-not-assigned)
```

### Expected behavior

no errors

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.5
Python 3.10.0 (tags/v3.10.0:b494f59, Oct  4 2021, 19:00:18) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

windows 10
